### ISSUE TITLE
connection reset by peer

### ISSUE THREAD

**Issue by fzong76** [hashicorp/terraform#24505]
```
terraform get terraform init -backend-config="storage_account_name=xxxxx" -backend-config="container_name=xxxx" -backend-config="key=xxx.tfstate" -backend-config="access_key=xxxxx" -get=false -reconfigure
Initializing the backend...

Successfully configured the backend "azurerm"! Terraform will automatically use this backend unless the backend configuration changes.

Error: Failed to get existing workspaces: Get https://xxx.blob.core.windows.net/xxxx?comp=list&prefix=xxxx.tfstateenv%3A&restype=container: read tcp 10.10.1.1:26822->4.8.1.4:443: read: connection reset by peer
```

I raised the issue to terraform-provider-azurerm https://github.com/terraform-providers/terraform-provider-azurerm/issues/6285.  And here is the reply from them

`This issue is related to the Azure Backend rather than the Azure Provider (which was fixed in #3908) - as such this issue needs to be opened on the Terraform Core repository. Unfortunately since these repositories are in separate Github orgs, at this time there's no way to transfer these and keep you subscribed to this issue - as such would you mind re-opening this issue on the Terraform Core repository?`

**Comment by Aedalus** [comment]
@fzong76 I seem to be hitting this issue as well, using terraform v0.12.19 with provider v1.44.0.

I'm curious if you ever found a workaround? My problem seems intermittent, and seems to have first appeared when I tried to use virtual endpoints to connect to blob storage.

Thanks!

**Comment by tombuildsstuff** [comment]
This should be fixed with 0.13.0 by switching to Giovanni - so I'm going to close this issue for the moment - but please let us know if the 0.13 release doesn't solve this and we'll take another look.
